src/equations/compressible_euler_plasma.jl

```diff
@@ -0,0 +1,323 @@
+    sqrt_rho_ll = sqrt(rho_ll)
+    sqrt_rho_rr = sqrt(rho_rr)
+    sum_sqrt_rho = sqrt_rho_ll + sqrt_rho_rr
+    vel_L = v1_ll
+    vel_R = v1_rr
+    vel_roe = (sqrt_rho_ll * vel_L + sqrt_rho_rr * vel_R) / sum_sqrt_rho
+    ekin_roe = 0.5f0 * vel_roe^2
+    H_ll = (rho_e_ll + p_ll) / rho_ll
+    H_rr = (rho_e_rr + p_rr) / rho_rr
+    H_roe = (sqrt_rho_ll * H_ll + sqrt_rho_rr * H_rr) / sum_sqrt_rho
+    c_roe = sqrt((gamma - 1) * (H_roe - ekin_roe))
+
+    Ssl = min(vel_L - c_ll, vel_roe - c_roe)
+    Ssr = max(vel_R + c_rr, vel_roe + c_roe)
+    sMu_L = Ssl - vel_L
+    sMu_R = Ssr - vel_R
+    if Ssl >= 0
+        f1 = f_ll[1]
+        f2 = f_ll[2]
+        f3 = f_ll[3]
+    elseif Ssr <= 0
+        f1 = f_rr[1]
+        f2 = f_rr[2]
+        f3 = f_rr[3]
+    else
+        SStar = (p_rr - p_ll + rho_ll * vel_L * sMu_L - rho_rr * vel_R * sMu_R) /
+                (rho_ll * sMu_L - rho_rr * sMu_R)
+        if Ssl <= 0 <= SStar
+            densStar = rho_ll * sMu_L / (Ssl - SStar)
+            enerStar = e_ll + (SStar - vel_L) * (SStar + p_ll / (rho_ll * sMu_L))
+            UStar1 = densStar
+            UStar2 = densStar * SStar
+            UStar3 = densStar * enerStar
+
+            f1 = f_ll[1] + Ssl * (UStar1 - rho_ll)
+            f2 = f_ll[2] + Ssl * (UStar2 - rho_v1_ll)
+            f3 = f_ll[3] + Ssl * (UStar3 - rho_e_ll)
+        else
+            densStar = rho_rr * sMu_R / (Ssr - SStar)
+            enerStar = e_rr + (SStar - vel_R) * (SStar + p_rr / (rho_rr * sMu_R))
+            UStar1 = densStar
+            UStar2 = densStar * SStar
+            UStar3 = densStar * enerStar
+
+            #end
+            f1 = f_rr[1] + Ssr * (UStar1 - rho_rr)
+            f2 = f_rr[2] + Ssr * (UStar2 - rho_v1_rr)
+            f3 = f_rr[3] + Ssr * (UStar3 - rho_e_rr)
+        end
+    end
+    return SVector(f1, f2, f3)
+end
+
+"""
+    flux_hllc(u_ll, u_rr, orientation, equations::CompressibleEulerPlasmaEquations1D)
+
+Computes the HLLC flux (HLL with Contact) for compressible Euler equations developed by E.F. Toro
+[Lecture slides](http://www.prague-sum.com/download/2012/Toro_2-HLLC-RiemannSolver.pdf)
+Signal speeds: [DOI: 10.1137/S1064827593260140](https://doi.org/10.1137/S1064827593260140)
+"""
+function flux_hllc(u_ll, u_rr, orientation::Integer, equations::CompressibleEulerPlasmaEquations1D)
+    flux_hllc_ion = single_species_flux_hllc(u_ll[1:3], u_rr[1:3], orientation, equations.gamma)
+    flux_hllc_el = single_species_flux_hllc(u_ll[4:6], u_rr[4:6], orientation, equations.gamma)
+    return vcat(flux_hllc_ion, flux_hllc_el)
+end
+
+@inline function max_abs_speeds(u, equations::CompressibleEulerPlasmaEquations1D)
+    rho_ion, v1_ion, p_ion, rho_el, v1_el, p_el = cons2prim(u, equations)
+    c_ion = sqrt(equations.gamma * p_ion / rho_ion)
+    c_el = sqrt(equations.gamma * p_el / rho_el)
+
+    return (abs(v1_ion) + c_ion, abs(v1_el) + c_el) # Is this the right way? I've seen multicomponent where (abs(v1) + c,) is returned
+end
```

Comment:
I am actually not sure what this is used for. What I would like to have, though, is `min_max_speed_davis` in order to be able to use the HLL solver.